### ISSUE TITLE
Settings management in a Multi-root Workspace context

### ISSUE THREAD

**Issue by FALLAI-Denis** [IBM/zopeneditor-about#55]
## Description of the enhancement requested

Despite what the documentation suggests, IBM Z Open Editor does not seem to manage its own settings at the folder level in a Multi-root Workspace context.
See https://ibm.github.io/zopeneditor-about/Docs/setting_propertygroup.html#quick-tips-for-setting-ibm-z-open-editor-property-groups-local-files-only

![Multi-root Workspace settings](https://user-images.githubusercontent.com/63861965/83775442-39476f00-a687-11ea-9a09-2d71badc4107.png)

When working in "Workspace" mode with several folders declared in this Workspace, (for example several GIT repositories), you should be able to manage certain IBM Z Open Editor settings for each root folder declared in the Workspace, because these settings can be contextual and not global.

The settings concerned by the "folder" level would be at least:
- zopeneditor.propertygroups: to reference mainframe PDS containing copybooks and includes or relative path in current folder hierarchy
- zopeneditor.zowe: to declare the profile to use to access the mainframe to search for copybooks and includes (all users using the folder (the GIT repository) must then use the same profile name)
- zopeneditor.cobol.enableUnreachableCodeWarnings: to activate or deactivate the functionality depending on the context

In general, any setting which is not necessarily global should be able to be managed at the folder level in the multi-root workspace operating mode.

**Comment by phaumer** [comment]
Yes, this is a valid enhancement request. Our docs might not be really clear about that, but Property Groups are window-level settings and not resource-level. So there can only be one definition of these at the moment, which needs to be defined on the top multi-root level. The lower level ones will be ignore. We definitely want to improve that situation.

**Comment by TimMSbg** [comment]
is there any timeframe for this enhancement? we really do need this aswell
thanks!
